### PR TITLE
A blog post for building the ocaml gpt library

### PR DESCRIPTION
**Pull request by PizieDust**
As part of my outreachy internship, I have written this blog article on how I developed the `ocaml-gpt` library.

### CODE REVIEW

**Comment by PizieDust** [comment]
cc @pitag-ha

**Review by PizieDust — COMMENTED** [review]
Thanks for the corrections @tmattio 👍

**Comment by sabine** [comment]
@tmattio @pitag-ha to me this implies the creation of an "Outreachy blog" that is hosted on OCaml.org. Is that the intention?

Currently, only the posts from the Opam blog are hosted on OCaml.org. I would be inclined to merge https://github.com/ocaml/ocaml.org/pull/1459 first, and then merge this blog post as part of a new "Outreachy blog" where Outreachy mentors, interns, etc. can publish blog posts.

**Comment by pitag-ha** [comment]
@PizieDust and I have discussed a few days ago that he'll set up his own medium blog, move this post there, and link to the post from here (just as done for other Outreachy posts). We should have communicated that here.

If instead you prefer the "Outrachy blog on ocaml.org" solution, let us know, and we'll discuss it. It has both up- and downsides.

**Comment by sabine** [comment]
Treating all the posts the same sounds reasonable to me. That's why I was wondering if you wanted to change how we handle the outreachy posts. Publishing elsewhere and adding a link is entirely fine

File: data/planet/-hosted_on_ocamlorg/building-the-ocaml-gpt-library.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+date: 2023-08-07T06:32:00-00:00
+preview_image:
+featured:
+authors:
+- Pizie Dust
+source:
+---
+
+## Introduction
+In the world today, software development often demands efficient management of storage devices. One of the key players in this arena is the GUID Partitioning Table (GPT). In this tutorial, we'll go over the steps of how the [`ocaml-gpt`](https://github.com/PizieDust/ocaml-gpt) library was developed and what this means for the OCaml ecosystem, especially in the context of the Mirage operating system, which is lacking in persistent storage capabilities. Through this `ocaml-gpt` library, developers will be able to seamlessly manage partitions in their block devices and disk images enabling enhanced control and reliability in storage management.
```
Comment (christinerose) [inline data/planet/-hosted_on_ocamlorg/building-the-ocaml-gpt-library.md]:
```suggestion
In the world today, software development often demands efficient management of storage devices. One of the key players in this arena is the GUID Partitioning Table (GPT). In this tutorial, we'll go over the steps of how the [`ocaml-gpt`](https://github.com/PizieDust/ocaml-gpt) library was developed and what this means for the OCaml ecosystem, especially in the context of MirageOS, which is lacking in persistent storage capabilities. Through this `ocaml-gpt` library, developers will be able to seamlessly manage partitions in their block devices and disk images enabling enhanced control and reliability in storage management. 
```

**Comment by christinerose** [comment]
Hi @PizieDust!

Great article! I went through and fixed any grammar/formatting discrepancies to match our style guide so as not to inundate you with comma, spelling, and capitalisation changes. 

I did have one suggestions about writing MirageOS like this rather than "Mirage operative system." The other instances where you had Mirage, I changed to MirageOS, as I've been instructed that's the official way to write it. :) 

~Maybe we can get this ready for publication this or next week!~
Oh! I see now it's not for publication on Tarides.com or OCaml.org? I think we published Outreachy posts on Tarides.com before. I'll double check with the Comms supervisor.

What do you think @sabine & @tmattio?

File: data/planet/-hosted_on_ocamlorg/building-the-ocaml-gpt-library.md (hunk 1 above)
Comment (PizieDust) [reply data/planet/-hosted_on_ocamlorg/building-the-ocaml-gpt-library.md]: Thank you for the corrections @christinerose

**Comment by PizieDust** [comment]
I created a medium account and will just port over the article on there and link to it from here, as most interns have done in the past.

https://medium.com/@playersrebirth/building-the-ocaml-gpt-library-650c00b4d72b

**Comment by christinerose** [comment]
I spoke with my team yesterday, and they like the idea of putting it on Tarides.com. We have published several intern blogs there before. Now that you've published it on Medium, I've checked in with my supervisor to see if we want to crosspost it or write a short introduction and link to it.

**Comment by sabine** [comment]
Thanks @PizieDust! I'll be taking it from here and merging. Nice blog post!

**Pull request merged**